src/query/src/promql/planner.rs

```diff
@@ -1425,6 +1457,127 @@ impl PromPlanner {
         Ok(exprs)
     }
 
+    /// Build expr for `label_replace` function
+    fn build_regexp_replace_label_expr(
+        other_input_exprs: &mut VecDeque<DfExpr>,
+        session_state: &SessionState,
+    ) -> Result<DfExpr> {
+        // label_replace(vector, dst_label, replacement, src_label, regex)
```

Comment:
What if `dst_label` has the same name as `src_label`?

Reply:
Good catch! Fixed in https://github.com/GreptimeTeam/greptimedb/pull/5153/commits/9750e5fc83f6eeb4b502db00581540c03cf720f0

It will replace the source label with the generated new label value.